Jenkinsfile

```diff
@@ -0,0 +1,214 @@
+#!groovy
+library 'jenkins-pipeline-shared'
+
+pipeline {
+    environment {
+        TEAM = "sbr"
+        MODULE_NAME = "zipkin-server"
+        MANIFEST_DIR = "resources"
+        DEV = "dev"
+        TEST = "test"
+        PROD = "beta"
+        FAILED_STAGE = "NONE"
+    }
+    options {
+        skipDefaultCheckout()
+        buildDiscarder(logRotator(numToKeepStr: '30', artifactNumToKeepStr: '30'))
+        timeout(time: 30, unit: 'MINUTES')
+        timestamps()
+    }
+    agent any
+    stages {
+        stage('Checkout') {
+            agent any
+            steps {
+                deleteDir()
+                checkout scm
+                script {
+                    def server = Artifactory.server 'art-p-01'
+                    def buildInfo = Artifactory.newBuildInfo()
+                    def downloadSpec = readFile 'resources/download.json'
+                    server.download spec: downloadSpec, buildInfo: buildInfo
+                }
+                sh "cp ${MODULE_NAME}/${MODULE_NAME}*.jar ${MODULE_NAME}.jar"
+            }
+            post {
+                success {
+                    stageSuccess()
+                }
+                failure {
+                    stageFailure()
+                }
+            }
+        }
+
+        stage('Test'){
+            agent any
+            steps {
+                script {
+                    def exists = fileExists "${MODULE_NAME}.jar"
+                    if (exists) {
+                        stageSuccess()
+                    } else {
+                        stageFailure()
+                        error("Cannot find file ${MODULE_NAME}.jar")
+                    }
+                }
+            }
+        }
+
+        stage('Deploy - DEV') {
+            agent any
+            when { branch 'master' }
+            environment {
+                DEV_ROUTE = "${DEV}-${TEAM}-${MODULE_NAME}"
+            }
+            steps {
+                milestone(1)
```

Comment:
Is it worth naming milestones to be a little clearer? `milestone(label: "begin deploy dev", ordinal: 1)`